Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -188,4 +188,162 @@ extension HTTP2ServerTransport.Config {
       Self(maxRequestPayloadSize: 4 * 1024 * 1024)
     }
   }
+
+  public struct TransportSecurity: Sendable {
+    private enum Wrapped {
+      case plaintext
+      case tls(TLS)
+    }
+
+    private let wrapped: Wrapped
+
+    /// This connection is plaintext: no encryption will take place.
+    public static let plaintext = Self(wrapped: .plaintext)
+
+    /// This connection will use TLS.
+    public static func tls(_ tls: TLS) -> Self {
+      Self(wrapped: .tls(tls))
+    }
+
+    /// Returns the TLS configuration, if the security has been set to ``tls(_:)``.
+    public var tlsConfig: TLS? {
+      switch wrapped {
+      case .plaintext:
+        return nil
+      case .tls(let config):
+        return config
+      }
+    }
```

Comment:
Do we need this? Not sure users would need to get this value out.

Reply:
I'm using it internally (across modules actually), so I've changed it to be `package`